### PR TITLE
Adds immutable to the dependencies

### PR DESCRIPTION
**Pull request by arcanis**
### This is a ...

- [x] Bug fix

### What's the background?

The `rc-editor-mention` package has a peer dependency on `immutable` which isn't satisfied by `antd`.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?

Probably not. It might cause Immutable to be included in the dependency tree one extra time, but given that the current behavior is unspecified it doesn't sound too bad.

> 2. What will say in changelog?

I don't think it needs to be described in the changelog? Or just a line to explain that immutable is now listed as a dependency.

> 3. Does this PR contains potential break change or other risk?

Nope.

### Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
It is a warning that won't affect actual functions.

#14542

**Comment by arcanis** [comment]
@afc163 I work on Yarn on a daily basis, it's not just a warning without consequences. Do you mind reopening this PR?

The main effect is that since package managers aren't aware of the peer dependency, whether it works or not depends entirely on the hoisting strategy. The hoisting strategy is **unspecified** and as such might operate differently between package managers, or even the same project with different.

In short: it is broken even if you don't see it. Please fix it 🙏

**Comment by afc163** [comment]
> The `rc-editor-mention` package has a peer dependency on `immutable` which isn't satisfied by `antd`.

The `rc-editor-mention` already had a dependency of `immutable` but not a peer dependency.

https://github.com/react-component/editor-mention/blob/master/package.json#L65

**Comment by arcanis** [comment]
Oh you're right - it did have a peer dependency on immutable [in 1.1.7](https://github.com/react-component/editor-mention/blob/1.1.7/package.json#L70) which disappeared in https://github.com/react-component/editor-mention/pull/31, hence my confusion.

It's all fine then, thanks for your help!
